src/components/DevelopmentGuide/V4MigrationGuide.mdx

```diff
@@ -0,0 +1,256 @@
+using Reservoir component `id`s in tests, styles, or elsewhere, you may need to update them.
+
+#### Ensure responsive designs are compatible with the new breakpoints.
+
+Our [breakpoints](../?path=/docs/style-guide-breakpoints--docs) have changed! We don’t
+expect this to be too disruptive, but it’s possible a page will not look as expected on
+smaller screen sizes and will need to be adjusted.
+
+<Flex
+  direction={{ base: 'column', md: 'row' }} 
+  gap="m"
+  margin="0 auto"
+  width="75%" 
+  sx={{
+    '& table': {
+        margin: 'auto', 
+        width: '100%'
+    }
+  }}
+>
+  <Box width="100%">
+    <Heading level="h5" mb={4} size="heading6" textAlign="center">Old Breakpoints</Heading>
+    {/* prettier-ignore */}
+    | Name | Value |
+    | :--- | :---- |
+    | Base | 0 |
+    | Mobile | 320px |
+    | Tablet portrait | 600px |
+    | Tablet landscape | 960px |
+    | Desktop | 1280px |
+  </Box>
+
+ <Box width="100%">
+    <Heading level="h5" mb={4} size="heading6" textAlign="center">New Breakpoints</Heading>
+    {/* prettier-ignore */}
+    | Name | Value |
+    | :--- | :---- |
+    | Small mobile (base) | 0 |
+    | Large mobile | 480px |
+    | Small tablet | 768px |
+    | Large tablet | 1024px |
+    | Desktop | 1280px |
+  </Box>
+</Flex>
+
+#### Rename `tableTextSize` to `fontSize`.
+
+We updated the `tableTextSize` prop in `Table` to `fontSize` to align more closely with the CSS property.
+
+## Non-Breaking Changes
+
+These updates enhance flexibility, stability, and developer experience without requiring changes to your existing code.
+
+#### A `useResponsiveSpacing` hook was introduced.
+
+[`useResponsiveSpacing`](../?path=/docs/hooks-useresponsivespacing--docs) exports spacing values to generate consistent layouts.
+We recommend its usage for custom layout and components.
+
+#### The `id` prop is now optional for all components.
+
+For components that require `id`s to meet accessibility requirements, one will automatically be generated using React's `useId`
+hook if none is passed.
+
+#### All components now include a default `data-testid` attribute on their top-level element.
+
+The `data-testid` value follows the pattern `ds-[componentName]` (e.g., `ds-accordion`, `ds-button`).
+
+#### Component prop interfaces have been significantly expanded to accept a wider range of values.
+
+This includes standard native HTML attributes (e.g., `aria-label`, `type` for inputs, `tabIndex`) and all Chakra UI style props
+(e.g., `p`, `m`, `bg`, `color`, `sx`).
+
+Refer to the dedicated note above each component's props table for details on how its types were extended.
+
+#### We're no longer throwing errors.
+
+Components like `Breadcrumbs`, `Heading`, and `Image` no longer throw errors (crashing your application) when missing required prop
+values.
+
+#### There are no warnings for invalid children.
+
+Components that expect specific child types (e.g., `RadioGroup` expecting `Radio` children) will no longer log warnings reducing
```

Comment:
I like the "reducing console noise...". From reading through this, this might be one update that gives a reason for the update.